src/test/java/org/terasology/genome/GeneIndexGenomeMapTest.java

```diff
@@ -64,7 +59,7 @@ public Integer apply(@Nullable String input) {
 
         GenomeDefinition genomeDefinition = new GenomeDefinition(breedingAlgorithm, genomeMap);
         GeneIndexGenomeMap genomeMap1 = (GeneIndexGenomeMap) genomeDefinition.getGenomeMap();
-        assertEquals(genomeMap1.propertyDefinitionMap.toString(), genomeMap.propertyDefinitionMap.toString());
+        assertEquals(genomeMap1.getPropertyDefinitionMap().toString(), genomeMap.getPropertyDefinitionMap().toString());
```

Comment:
Hm, my gut feeling says that comparing to `Map`s by the String representation is not really a best practice... Is JUnit capable of comparing two maps in some way? Otherwise, instead of comparing the internal representations, maybe it makes sense to implement `GeneIndexGenomeMap::equals` so that you can do 
```java
assertEquals(genomeMap1, genomeMap);
```

Would comparing `GeneIndexGenomeMap` be a common operation in general, or just for testing?

Reply:
JUnit can use assertThat statements to check if a map contains certain elements. This can probably be used for a better test here. Since comparing such maps will probably never be used anywhere apart from testing, it might not be worth implementing the equals method there.